README.md

```diff
@@ -74,16 +74,30 @@ of what's defined in the configuration file.
 
 ## Compiling
 
-Download the repository and its dependencies:
+1. Install `idf.py`
+
+You will first need to install the 
+[Espressif IoT Development Framework](https://github.com/espressif/esp-idf) 
+to get the `idf.py` file. The [Installation Instructions](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/get-started/linux-macos-setup.html) have all of the details. Make sure to follow ALL the steps, up to and including step 4 where you set up the tools and the `get_idf` alias.
+
+2. Download the repository and its dependencies:
+
 ```bash
 git clone --recursive https://github.com/shmuelzon/esp32-ble2mqtt
 ```
+
+3. Modify the config.json and flash
+
 Modify the [configuration file](#configuration) to fit your environment, build
 and flash (make sure to modify the serial device your ESP32 is connected to):
 ```bash
 idf.py flash -p /dev/ttyS1
 ```
 
+You can get the TTY of your device (on linux) by running `dmesg | grep tty` and
+then plugging in your device, and running the command again. You are looking for
+a line similar to: `usb 3-2: cp210x converter now attached to ttyUSB0`.
+
```

Comment:
This is very specific to (a) Linux and (b) CP210x USB to Serial adapters.
Actually, newer versions of `idf.py` try to detect the serial device automatically, so maybe we can just leave this out?

Reply:
Sure thing, removed this whole block.